Made the storage object work asynchronously.

andrayantelo opened the pull request.

andrayantelo committed:
Made a public directory to serve from.

andrayantelo committed:
Removed firebase initialization code.

andrayantelo committed:
Rewrote allCalendarIds to return a promise. `

andrayantelo committed:
Updated the storage methods.
All the storage methos now return promises.

andrayantelo committed:
Modified the checkit methods that use storage obj
Used Promise methods like then and catch where
necessary.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -677,14 +700,26 @@ var LocalCalendarStorage = function(params) {
-        return prefix + "_" + id;
-        
+        var key = prefix + "_" + id;
+        console.log("toKey("+id+") -> " + key);
```
You can remove this log now

johntyree commented on public/calendar_reboot.js:
```diff
@@ -677,14 +700,26 @@ var LocalCalendarStorage = function(params) {
+                resolve(allCalendarIds);
+            }
+            else {
+                console.log(allCalendarIds);
```
You can remove this too. We have already verified that it is `null`.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -694,10 +729,17 @@ var LocalCalendarStorage = function(params) {
         storeInLocalStorage(toKey(calendarObj.state.uniqueId), calendarObj.state);
```
This is independent of the `getAllCalendarIds` part and should be wrapped in its own promise. Then you can compose the two together using the `Promise.*` functions and return *that*. The `save()` call isn't really "complete" until both the `state` is stored and the `allCalendarIds` are updated.

johntyree replied:
When we switch to firebase, I assume it will be possible to make several changes to the database together in a single call, but this is good practice now.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -710,38 +752,60 @@ var LocalCalendarStorage = function(params) {
+                // Remove active status from the calendar
+                removeFromLocalStorage(toKey(current_active_calendar));
+            }.bind(this))
+            .catch(function () {
```
Is this meant to imply that no matter what happens (i.e. network problems, missing data, etc) you want to ignore it entirely without telling anyone?

andrayantelo replied:
No, it just means I forgot to type something in there.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -50,35 +50,52 @@ function CheckIt() {
+        
+        .catch(function (value) {
+            console.log("No calendars in storage.");
+            return;
```
`return` is unneeded here.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -50,35 +50,52 @@ function CheckIt() {
+            
+            for (var key in allCalendarIds) {
+                if (allCalendarIds.hasOwnProperty(key)) {
+                this.addCalendarToDropdown(key, allCalendarIds[key]);
```
indent

johntyree commented on public/calendar_reboot.js:
```diff
@@ -50,35 +50,52 @@ function CheckIt() {
+                }
+            }
+        }.bind(this))
+        
```
It would be best not to put whitespace randomly everywhere because it tricks the reader into thinking things are unrelated. In this case, `.catch()` is being called on the return value of `.then()`, but it kind of looks like it stands alone somehow.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -50,35 +50,52 @@ function CheckIt() {
+               }.bind(this))
+               
+               .catch(function() {
+                   this.showBuildMenu();
```
If we got here then that means that our `activeCalendar` doesn't exist. This is a serious error and means the data is corrupted somehow. You should at a minimum `console.error()` it and maybe even try to correct the data.

johntyree commented on public/calendar_reboot.js:
```diff
@@ -136,12 +153,16 @@ CheckIt.prototype.loadFromDropdown = function( event ) {
+            var calendar = new Calendar(state, this);
+            this.displayCalendar(calendar);
+            this.hideBuildMenu(); 
+            }.bind(this))
```
indentation

johntyree replied:
also whitespace

johntyree commented on public/calendar_reboot.js:
```diff
@@ -154,18 +175,20 @@ CheckIt.prototype.deleteCalendar = function() {
+                //clear the page
+                this.clearPage();
+                this.showBuildMenu(); 
+                }.bind(this))
```
indentation and ws

andrayantelo committed:
Indent, removed white space, added removeActive
Added the removeActive method to store object so that
if for some reason we try to access current_active_calendar
from the storage and it returns null (and not a calendar state)
then we can remove that key from storage.

andrayantelo committed:
Combined two promises into one, added a print statement

johntyree commented on public/calendar_reboot.js:
```diff
@@ -710,38 +752,69 @@ var LocalCalendarStorage = function(params) {
+                removeFromLocalStorage(toKey(current_active_calendar));
+            }.bind(this))
+            .catch(function () {
+                console.log("Unable to remove calendar);
```
Missing ending quote

andrayantelo committed:
Added end quote
